docs/book/src/tasks/experimental-features/machineset-preflight-checks.md

```diff
@@ -1,12 +1,64 @@
 # Experimental Feature: MachineSetPreflightChecks (alpha)
 
-The `MachineSetPreflightChecks` feature can provide additional safety while creating new Machines for a MachineSet.
+The `MachineSetPreflightChecks` feature can provide additional safety while creating new Machines and remediating existing unhealthy Machines of a MachineSet.
+
+When a MachineSet creates machines under certain circumstances, the operation fails or leads to a new machine that will be deleted and recreated in a short timeframe,
+leading to unwanted Machine churn. Some of these circumstances include, but not limited to, creating a new Machine when Kubernetes version skew could be violated or 
+joining a Machine when the Control Plane is upgrading leading to failure because of unclear kube-apiserver version. 
```

Comment:
```suggestion
joining a Machine when the Control Plane is upgrading leading to failure because of mixed kube-apiserver version or due to the cluster load balancer delays in adapting to the changes. 
```